### ISSUE TITLE
ability to use custom filename in sendDocument etc.

### ISSUE THREAD

**Issue by ergoz** [leandrotoledo/python-telegram-bot#56]
I have a code that sends txt file, i need use tmp files that have bad names, when i use 

```python
temp_file = tempfile.NamedTemporaryFile(delete=False, mode='w')
tmp_filename = temp_file.name
temp_file.write(build_log)
temp_file.close()
document = open(tmp_filename, 'rb')
self.BOT.sendDocument(chat_id=chat_id, document=document)
document.close()
os.unlink(tmp_filename)
```

in telegram file have dynamicly generated tmp name (like tempsx8eth). How to set custom name? 
As i understand we need to refactor ```class InputFile(object)``` on line 70
```python
self.filename = os.path.basename(self.input_file.name)
```
I've seen 1-2 bots that have ability to change sending name and it is very usefull.

**Comment by leandrotoledo** [comment]
Hey, thanks for the bug report. If I got it right it does send without any problems you just want to have their names set before uploading, is that right? If so, we could refactor send* methods (the ones sending InputFiles) adding an extra custom_filename arg, would that work?

**Comment by ergoz** [comment]
Yes, You are right, I want to set custom name. Currently sendDocument code uses this line:
```python
self.filename = os.path.basename(self.input_file.name)
```
and when code generates headers:
```python
form.extend([
            form_boundary,
            'Content-Disposition: form-data; name="%s"; filename="%s"' % (
                self.input_name, self.filename
                ),
            'Content-Type: %s' % self.mimetype,
            '',
            self.input_file_content
        ])
```
it uses this name, that get from file path. If you will add arg for custom file name for all message functions that uses InputFile class it will be great!

I am a newbie in python, and i can't help with refactoring, but i think you will can make it in few lines of code. :)
Big thanks for this awesome library, i already create 3 bots with this lib. 
And sorry for bad English :)

**Comment by leandrotoledo** [comment]
Thank -you for helping it to get even better. Stay tuned, I'll bump this bug again once pushed to a pip package.

**Comment by ergoz** [comment]
Thank you very much! I will be waiting for next release :+1:

**Comment by ergoz** [comment]
I made it with ugly code :)

i've added param for sendDocument, sendVideo, etc. with name filename and added filename key for data dictionary, for example like this:
```python
@log
    @message
    def sendDocument(self,
                     chat_id,
                     document,
                     filename=None,
                     **kwargs):
        """Use this method to send general files.

        Args:
          chat_id:
            Unique identifier for the message recipient - User or GroupChat id.
          document:
            File to send. You can either pass a file_id as String to resend a
            file that is already on the Telegram servers, or upload a new file
            using multipart/form-data.
          filename:
            File name that shows in telegram message (it is usefull when you
            send file generated by temp module for example).
          reply_to_message_id:
            If the message is a reply, ID of the original message. [Optional]
          reply_markup:
            Additional interface options. A JSON-serialized object for a
            custom reply keyboard, instructions to hide keyboard or to force a
            reply from the user. [Optional]

        Returns:
          A telegram.Message instance representing the message posted.
        """

        url = '%s/sendDocument' % self.base_url

        data = {'chat_id': chat_id,
                'document': document,
                'filename': filename
        }

        return url, data
```

after this I  added if statment to set ```self.filename``` in InputFile class:
```python
if isinstance(self.input_file, file):
            self.input_file_content = self.input_file.read()
            if self.data.has_key('filename') and self.data['filename'] is not None:
                self.filename = self.data['filename']
            else:
                self.filename = os.path.basename(self.input_file.name)
            print self.input_file.name
            print self.filename
            self.mimetype = mimetypes.guess_type(self.filename)[0] or \
                DEFAULT_MIME_TYPE
```
I know that it is can be ugly :) but it's work for me :)

**Comment by leandrotoledo** [comment]
Cool! Would you mind to push as a PR into testing branch?

**Comment by ergoz** [comment]
and i don't know will this fix work on python 3+ ^_^ i tested it only on python 2.7.8

**Comment by ergoz** [comment]
No problem. I will push it in few minutes :)

**Comment by ergoz** [comment]
I finished. PR created. #57

**Comment by leandrotoledo** [comment]
It doesn't seem to make any difference on client side for:
- Sticker
- Voice
- Video

Can you confirm? Otherwise we can stick with just for Photo and Document classes.

**Comment by ergoz** [comment]
I will test all options tomorrow morning. I think it can work with stickers and etc, but it can be not needed in it :)

**Comment by leandrotoledo** [comment]
PR #59 to keep it just for sendDocument.

**Comment by leandrotoledo** [comment]
This feature has been released on v2.8.3, please don't forget to:
`pip install python-telegram-bot --upgrade`
